scripts/download.ts

```diff
@@ -0,0 +1,64 @@
+import { createReadStream, createWriteStream, promises } from 'fs';
+import { resolve } from 'url';
+import { join } from 'path';
+import { get } from 'https';
+import { Extract } from 'unzipper';
+
+import { Target, targets } from './targets';
+
+const { mkdir, unlink } = promises;
+
+const revision = process.argv[2];
+const baseUrl = 'https://www.googleapis.com/download/storage/v1/b/chromium-browser-snapshots/o/';
+const getUrl = (target: Target) => resolve(
+  baseUrl,
+  `${target.prefix}%2F${revision}%2F${target.name}.zip?alt=media`,
+);
+
+const baseDirectory = join(process.cwd(), 'chromium');
+const getArchivePath = (target: Target) => join(baseDirectory, `${target.name}.zip`);
+
+const download = (from: string, to: string) => new Promise<void>(resolve => {
+  const stream = createWriteStream(to);
+
+  get(from, response => {
+    response.pipe(stream);
+    response.on('end', () => {
+      stream.end();
+      resolve();
+    });
+  });
+});
+
+const unzip = (from: string, to: string) => new Promise((resolve, reject) => {
+  const stream = createReadStream(from);
+  const extract = Extract({ path: to });
+
+  extract.on('error', reject);
+  extract.once('close', resolve);
+  stream.pipe(extract);
+});
+
+(async () => {
+  try {
+    await mkdir(baseDirectory, {
+      recursive: true,
+    });
+  } catch (error) {
+    if (error?.code !== 'EEXIST') {
+      throw error;
+    }
+  }
```

Comment:
ここが (他コメントで書いている) 共通化したいとこ
`mkdir` に渡すオプション引数とかも転送できるようにしておきたい

Reply:
これも Chromium ダウンロード用スクリプトなのでできるだけ切り離したい

Reply:
じゃあ、これは共通化した後に出そうと思っていたものなんですが、

```suggestion
  await mkdir(baseDirectory, {
    recursive: true,
  }).catch((error) => {
    if (error?.code !== 'EEXIST') {
      throw error;
    }
  });
```

こういうふうに書けると思います

Reply:
あっ……たしかに全体で Promise にできる……
直します

Reply:
cc23792